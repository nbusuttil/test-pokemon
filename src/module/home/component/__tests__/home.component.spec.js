import React from 'react';
import { shallow } from 'enzyme';

import Home from '../home.component';

describe('Home component', () => {
  let props;

  beforeEach(() => {
    props = {
      loadPokemons: jest.fn(),
      openModal: jest.fn(),
      loadPokemonsWithPagination: jest.fn(),
      addItem: jest.fn(),
      pokemons: [
        {
          id: 2,
          price: 7,
          quantity: 0,
          name: 'spearow',
          url: 'https://pokeapi.co/api/v2/pokemon/21/'
        },
        {
          id: 1,
          price: 4,
          quantity: 0,
          name: 'fearow',
          url: 'https://pokeapi.co/api/v2/pokemon/22/'
        }
      ],
      filteredBy: '',
      modal: false,
      nextUrl: ''
    };
  });
  const getWrapper = () => shallow(<Home {...props} />);

  it('render without crashing', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
