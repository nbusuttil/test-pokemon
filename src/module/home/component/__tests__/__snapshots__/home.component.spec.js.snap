// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component render without crashing 1`] = `
<div
  className="home"
>
  <div
    className="home__filter"
  >
    <p
      className="home__filter-label"
    >
      Choose order
    </p>
    <select
      className="home__filter-select"
      onChange={[Function]}
      value=""
    />
  </div>
  <Card
    action={[MockFunction]}
    key="2"
    openModal={[MockFunction]}
    pokemon={
      Object {
        "id": 2,
        "name": "spearow",
        "price": 7,
        "quantity": 0,
        "url": "https://pokeapi.co/api/v2/pokemon/21/",
      }
    }
    titleAction="Add to cart"
  />
  <Card
    action={[MockFunction]}
    key="1"
    openModal={[MockFunction]}
    pokemon={
      Object {
        "id": 1,
        "name": "fearow",
        "price": 4,
        "quantity": 0,
        "url": "https://pokeapi.co/api/v2/pokemon/22/",
      }
    }
    titleAction="Add to cart"
  />
  <div
    className="home__action"
  >
    <button
      onClick={[Function]}
    >
      <img
        alt="more"
        src="more.svg"
      />
    </button>
  </div>
</div>
`;
